### PR TITLE
[Feature] Add onHold state for remote participant with UI/UX update

### PR DESCRIPTION
**Pull request by AmyL219**
## Purpose

* Add onHold state for remote participant with UI/UX update
Simulator Screen Shot - iPhone 11 - 2022-05-27 at 10 12 03

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Service/Calling/Extension/ACSParticipantStateExtension.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        case .hold:
+            return .hold
+        case .inLobby:
+            return .inLobby
+        case .disconnected:
+            return .disconnected
+        case .ringing:
+            return .ringing
+        default:
+            return .idle
```
Comment (kevinyulu) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Service/Calling/Extension/ACSParticipantStateExtension.swift:28]: out of curiosity, what the rule from SDK that would determine if a participant is `.idle` vs `.connected`?

Comment (kevinyulu) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Service/Calling/Extension/ACSParticipantStateExtension.swift:28]: also would we default the state to .connected?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/Grid/Cell/ParticipantGridCellViewModel.swift [hunk 2]
```diff
@@ -76,6 +78,12 @@ class ParticipantGridCellViewModel: ObservableObject, Identifiable {
         if self.isMuted != participantModel.isMuted {
             self.isMuted = participantModel.isMuted
         }
+
+        let isOnHold = participantModel.status == .hold
+
+        if self.isHold != isOnHold {
```
Comment (kevinyulu) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/Grid/Cell/ParticipantGridCellViewModel.swift:84]: Tested with having a teams call with 2 participants: (Kevin Guo (host) and User 2 (guest)) with the following two scenarios:
Scenario 1 **Kevin Guo is on hold**: (tested OK) host will show on hold UX in participant grid and participant list drawer. 

Scenario 2 **User 2 is on hold**: when this case happens, the app shows an on hold status on the host _Kevin Guo_ in the participant drawer. Should the guest User 2 be labeled as on hold instead? Screenshot attached
IMG_0090

**Review by kevinyulu — COMMENTED** [review]
tested the app in iPhone XR using Teams and left a couple of comments on some test cases.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/Grid/Cell/ParticipantGridCellViewModel.swift (hunk 2 above)
Comment (AmyL219) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/Grid/Cell/ParticipantGridCellViewModel.swift:84]: Yeah, the CallingSDK updates all remote participants with `.hold` state. Let me dismiss the participant list when the local user is update to `onHold` state which would be consistent with Android side

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Service/Calling/Extension/ACSParticipantStateExtension.swift (hunk 1 above)
Comment (AmyL219) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Service/Calling/Extension/ACSParticipantStateExtension.swift:28]: I think `.idle` is more equal to the localUserState `.none`. I think we would only update the remote participant state to .connected when he has the .connected status from CallingSDK

**Pull request merged**